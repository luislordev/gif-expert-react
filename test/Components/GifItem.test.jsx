import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/Components/GifItem';

describe('Gif item component', () => {
  const title = 'Kenobi';
  const url = 'https://example.com/kenobi.gif';

  test('Match with Snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test('Show image with url and alt text', () => {
    render(<GifItem title={title} url={url} />);

    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Show title in the component', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
