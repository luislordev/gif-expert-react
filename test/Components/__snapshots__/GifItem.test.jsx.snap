// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif item component Match with Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Kenobi"
      src="https://example.com/kenobi.gif"
    />
    <p>
      Kenobi
    </p>
  </div>
</div>
`;
